src/storage/mvcc/reader/scanner/mod.rs

```diff
@@ -97,25 +99,45 @@ impl<S: Snapshot> ScannerBuilder<S> {
             )))
         }
     }
+
+    pub fn build_entry_scanner(mut self) -> Result<Scanner<S>> {
+        let lock_cursor = self.create_cf_cursor(CF_LOCK)?;
+        let write_cursor = self.create_cf_cursor(CF_WRITE)?;
+        Ok(Scanner::Entry(EntryScanner::new(
+            self.0,
+            lock_cursor,
+            write_cursor,
+        )?))
+    }
 }
 
 pub enum Scanner<S: Snapshot> {
     Forward(ForwardScanner<S>),
     Backward(BackwardScanner<S>),
+    Entry(EntryScanner<S>),
 }
 
 impl<S: Snapshot> StoreScanner for Scanner<S> {
     fn next(&mut self) -> TxnResult<Option<(Key, Value)>> {
         match self {
             Scanner::Forward(scanner) => Ok(scanner.read_next()?),
             Scanner::Backward(scanner) => Ok(scanner.read_next()?),
+            Scanner::Entry(_) => Err(box_err!("unimplemented")),
```

Comment:
Can we extract EntryScanner from Scanner, to keep Scanner purely working on KVs, considering that all places using this Scanner will not use the EntryScanner even in future? Besides, we will heavily modify Scanners' interface in MVCC batch scan (in order to reduce copy), you may not want Backup/Restore being affected by this as well.

Reply:
Okay, it may add extra code.